tests/engine/test_filter_evttype_resolver.cpp

```diff
@@ -234,4 +244,127 @@ TEST_CASE("Should find event types from filter", "[rule_loader]")
         auto f = compile("not (not evt.type=openat)");
         compare_evttypes(std::move(f), openat_only);
     }
+
+    SECTION("not_ne_equivalence")
+    {
+        compare_filters_evttypes("not evt.type=mmap", "evt.type!=mmap");
+        compare_filters_evttypes("not proc.name=cat", "proc.name!=cat");
+    }
+
+    // see: https://github.com/falcosecurity/libs/pull/854#issuecomment-1411151732
+    SECTION("libs #854")
+    {
+        std::set<uint16_t> empty;
+        compare_filters_evttypes(
+            "evt.type in (connect, execve, accept, mmap, container) and not (proc.name=cat and evt.type=mmap)",
+            "evt.type in (accept, connect, container, execve, mmap)");
+        compare_evttypes(compile("(evt.type=mmap and not evt.type=mmap)"), empty);
+    }
+
+    SECTION("boolean algebra")
```

Comment:
Adding this test section allows us to assert that the set of event types correctly follows the boolean algebra semantics.